railgun/manager/run.go

```diff
@@ -175,6 +176,15 @@ func Run(ctx context.Context, c *config.Config) error {
 				Proxy:  prx,
 			},
 		},
+		{
+			IsEnabled: &alwaysEnabled,
```

Comment:
To be more consistent, a SecreteEnablement field can be defined within controller config structure as all other controller do. Please use that

Reply:
The way we load Secrets is most like Endpoints, which [aren't independent](https://github.com/Kong/kubernetes-ingress-controller/blob/2ce512d67980a29fa6fea73127b6e9037912271e/railgun/manager/run.go#L171-L177) (they share enablement with Services).

Broadly, we load Secrets from other objects, e.g. to provide a certificate for an Ingress, configuration for a plugin, etc. They're used widely across a variety of types, however (versus Endpoints being specific to Services), so there are a number of controllers that potentially depend on them. The "proper" condition would be something like:

```
IsEnabled: &c.ServiceEnabled || &c.IngressNetV1Enabled || &c.IngressNetV1beta1Enabled || &c.IngressExtV1beta1Enabled || &c.TCPIngressEnabled || &c.KongClusterPluginEnabled || &c.KongPluginEnabled || &c.KongConsumerEnabled ...
```
Practically speaking, if you enable almost anything, you also enable the Secret controller. I can only think of very contrived configurations that shouldn't exist in practice where it's not a dependency (e.g. enabling only the KongIngress controller), so simply enabling it always instead of writing out the complete condition seemed appropriate. Does that reasoning make sense/do you still think this would need to change?

Reply:
hmm... understand. thanks for clarification. Please go as it is.